src/Residue.jl

```diff
@@ -450,6 +450,7 @@ to the constructor with the same base ring $R$ and element $a$. A modulus
 of zero is not supported and throws an exception.
 """
 function residue_ring(R::Ring, a::RingElement; cached::Bool = true)
+   @req !is_trivial(R) "Zero rings are currently not supported as base ring."
```

Comment:
Since `residue_ring` can already be a zero ring, I think it is kinda OK to allow a zero ring as base ring? OTOH I also see no strong *need* to support a zero ring as base ring. But perhaps it allows to make some code more uniform... In any case, we could probably handle it with some code like this:
```suggestion
   if is_trivial(R)
     a = one(R)  # base ring is trivial, so we might as well factor out 1
   end
```
Then again: if `R` is a zero ring, won't the next code line trigger anyway, which throws an error if `iszero(A)` is true? If that's a case, we may as well keep this line and strengthen it by removing `currently`

Reply:
I would keep the error as provided by the PR for now (with the "currently").

I think the line below with the reference to the C library is a bit misleading, as it is not relevant.